src/GitHub.App/Services/RepositoryCloneService.cs

```diff
@@ -223,6 +223,7 @@ bool IsSolutionInRepository(string repositoryPath)
             }
             catch (Exception ex)
             {
+                operatingSystem.Directory.DeleteDirectory(repositoryPath);
                 log.Error(ex, "Could not clone {CloneUrl} to {Path}", cloneUrl, repositoryPath);
```

Comment:
Maybe we should move the `log.Error(...)` to above `DeleteDirectory` in case it also throws?